src/components/Presentation/index.jsx

```diff
@@ -1,93 +1,174 @@
 import React from 'react';
-import Toolbar from 'components/Toolbar';
-import Button from 'components/Button';
-import {
-  Grid,
-} from '@material-ui/core';
-import { 
-  connect 
-} from 'react-redux';
-import {
-  compose
-} from 'redux';
-import withDragDrop from 'lib/withDragDrop';
 import { fabric } from 'fabric';
+import { useDrop } from 'react-dnd';
+import Canvas from 'components/Canvas';
+import Editor from 'components/Editor';
+import Slidebar from 'components/Slidebar';
+import Header from 'components/Header';
+import Editormenu from 'components/Editormenu';
+import Welcome from 'components/Welcome';
+import { Grid } from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+import { EditorContextProvider } from 'components/Editor/context';
+import { SlideContextProvider } from 'components/Slide/context';
+import { handleKeyboardShortcuts } from 'events/index';
 
 
-class Presentation extends React.PureComponent {
-  constructor (props) {
-    super(props);
-    this.state = {
-      activeSlide: null,
-    }
-    this.canvasEl = React.createRef();
-    this.canvas = null;
+let canvas = null;
+let hiddenCanvas = null;
+
+const useStyles = makeStyles(theme => ({
+  height: {
+    height: '100%'
+  },
+  grow: {
+    flexGrow: '1'
+  },
+  overflow: {
+    overflow: 'auto'
+  },
+  canvasContainer: {
+    boxSizing: 'border-box',
+    height: '100%',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    backgroundColor: theme.palette.grey['100']
+  },
+  padding: {
+    padding: theme.spacing(1)
+  },
+  slideBarContainer: {
+    backgroundColor: theme.palette.grey['100']
+  }
+}));
+
+const Presentation = ({
+  slides,
+  activeSlide,
+  setActiveSlide,
+}) => {
+  const classes = useStyles();
+  const canvasEl = React.createRef(null)
+  const hiddenCanvasEl = React.createRef(null)
+
+  const canvasObj = () => {
+    return canvas
   }
-  componentWillMount() {
-    this.canvas = new fabric.Canvas(this.canvasEl.current);
+
+  const hiddenCanvasObj = () => {
+    return hiddenCanvas
   }
-  componentWillReceiveProps(nextProps) {
-    
+
+  const updateSlideWithNewResolution = (canvasObjects, slideWidth) => {
+    if (canvas.width != slideWidth) {
```

Comment:
If strict comparison will work here, we should use it. If we do a shallow comparison like we are now, there are some instances in which it will fail